HPS/Heartland/Model/Payment.php

```diff
@@ -310,8 +302,9 @@ function validate()
             $errorMsg = __('This credit card type is not allowed for this payment method.');
         }*/
         // \HPS\Heartland\Model\Payment::getToken
-        if (!$this->getToken(new \HpsTokenData)) {
-            $errorMsg = __('No valid token.');
+        $suToken = $this->getToken(new \HpsTokenData);       
+        if(empty($suToken->tokenValue)){
```

Comment:
getToken will always return the HpsTokenData object , even there is no token value. so, the $suToken will not be null. Anyway there is no need to check $suToken variable, this "empty" method will check both $suToken and tokenValue empty or not